### PR TITLE
Correction workflow: As a SD admin I want to have a config to introduce a corrections workflow. [SDCP-390]

### PR DESCRIPTION
**Pull request by GyanP**

### CODE REVIEW

**Comment by petrjasek** [comment]
I wonder why we need those `correction_of` and `corrected_by` fields?

**Comment by GyanP** [comment]
> I wonder why we need those `correction_of` and `corrected_by` fields?

We need both of these for making a link between the **correction** and **being corrected** articles because we have an action button **cancel correction** which is used to reset the article to its previous state.

**Comment by petrjasek** [comment]
yep but the guid should be the same for both, you can only correct single item, not any other version of it right?

File: apps/publish/content/correct.py [hunk 1]
```diff
@@ -78,6 +80,22 @@ def set_state(self, original, updates):
         else:
             super().set_state(original, updates)
 
+    def change_being_corrected_to_published(self, updates, original):
+        if app.config.get('CORRECTIONS_WORKFLOW') and original.get('state') == 'correction':
+            publish_service = get_resource_service('published')
+            being_corrected_articles = publish_service.find({
```
Comment (petrjasek) [inline apps/publish/content/correct.py:86]: is it actually possible to have more than 1? if not using `find_one` would be better to make it explicit

Comment (petrjasek) [reply apps/publish/content/correct.py]: I might have been wrong here, if it's published collection there might be more items with same guid, that would be the case in archive

**Review by petrjasek — APPROVED** [review]

**Pull request merged**